.github/workflows/deploy-production.yaml

```diff
@@ -0,0 +1,53 @@
+name: Deploy to Cloudflare Pages on Release
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  deploy-production:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      deployments: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Node & PNPM
+        uses: pnpm/action-setup@v3
+        with:
+          version: 10.19.0
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'pnpm'
+          cache-dependency-path: docs/pnpm-lock.yaml
+
+      - name: Install Dependencies
+        run: |
+          cd docs
+          pnpm install
+          pnpm exec playwright install --with-deps
+
+      - name: Build Site
+        run: |
+          cd docs
+          pnpm run build
+
+      - name: Run E2E Tests
+        run: |
+          cd docs
+          pnpm run test
+
+      - name: Deploy to Cloudflare Pages
+        uses: cloudflare/pages-action@v1
+        with:
+          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+          projectName: supakone-me
+          directory: docs/dist
+          gitHubToken: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Do not build a separate file for deploy-production here since it work almost like a pr-preview.yaml 

so I Think you should merge these 2 file into one and using condiftion if release deploy to cloudfare if it a PR only ene to end test (then release do not end to end test)